beacon-chain/verification/lazy_head_state.go

```diff
@@ -1,63 +1,7 @@
 package verification
 
-import (
-	"context"
-
-	"github.com/OffchainLabs/prysm/v6/beacon-chain/state"
-	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
-)
-
-// HeadStateGetter is a function type that retrieves a HeadStateProvider.
-// This allows for lazy initialization of the head state provider.
-type HeadStateGetter func() (HeadStateProvider, error)
-
-// LazyHeadStateProvider wraps a HeadStateGetter to provide lazy access to head state methods.
-// This is useful when the underlying head state provider (e.g., blockchain service) is not
-// available at construction time but will be available when methods are called.
 type LazyHeadStateProvider struct {
-	getter HeadStateGetter
+	HeadStateProvider
```

Comment:
Can't we just then delete the file then? Also can't then just use `HeadStateProvider` instead of wrapping it in `LazyHeadStatePRovider`?

Reply:
yeah, makes sense to delete imo